mule-http-mock-extension/src/main/java/org/mule/test/extensions/http/mock/internal/server/HTTPMockServer.java

```diff
@@ -6,54 +6,174 @@
+  public class StubRemover {
 
-    private final StubMapping stubMapping;
-    private final WireMockServer wireMockServer;
+    private final String path;
 
-    public StubRemover(StubMapping stubMapping, WireMockServer wireMockServer) {
-      this.stubMapping = stubMapping;
-      this.wireMockServer = wireMockServer;
+    public StubRemover(String path) {
+      this.path = path;
     }
 
     public void removeStub() {
-      wireMockServer.removeStubMapping(stubMapping);
+      pathToCallback.remove(path);
+    }
+  }
+
+  private class MockServlet extends HttpServlet {
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+      handle(req, resp);
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+      handle(req, resp);
+    }
+
+    private void handle(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+      String path = req.getRequestURI();
+      SourceCallback<InputStream, HTTPMockRequestAttributes> callback =
+          (SourceCallback<InputStream, HTTPMockRequestAttributes>) pathToCallback.get(path);
+
+      if (callback == null) {
+        resp.setStatus(HttpServletResponse.SC_NOT_FOUND);
+        resp.getWriter().write("Not Found");
+        return;
+      }
+
+      byte[] requestBody = toByteArray(req.getInputStream());
+      HTTPMockServerResponse mockResponse = DelegateToFlowTransformer.delegate(callback, requestBody);
+
+      byte[] responseBody = mockResponse.getBody() != null ? toByteArray(mockResponse.getBody().getValue()) : new byte[0];
+
+      resp.setStatus(mockResponse.getStatusCode());
+      resp.setContentLength(responseBody.length);
+      resp.setCharacterEncoding("UTF-8");
+      resp.setContentType("application/json");
+
+      mockResponse.getHeaders().entrySet().forEach(entry -> resp.addHeader(entry.getKey(), entry.getValue()));
+      try (OutputStream os = resp.getOutputStream()) {
+        os.write(responseBody);
+        os.flush();
+      }
+    }
+
+    private byte[] toByteArray(InputStream input) throws IOException {
+      ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+      byte[] data = new byte[1024];
+      int nRead;
+      while ((nRead = input.read(data, 0, data.length)) != -1) {
+        buffer.write(data, 0, nRead);
+      }
+      return buffer.toByteArray();
+    }
+  }
+
+  private class ExpectContinueServlet extends HttpServlet {
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+      handle(req, resp);
+    }
+
+    private void handle(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+      String path = req.getRequestURI();
+      SourceCallback<InputStream, HTTPMockRequestAttributes> callback =
+          (SourceCallback<InputStream, HTTPMockRequestAttributes>) pathToCallback.get(path);
+
+      long expectHeaderReceived = 0;
```

Comment:
What do these numbers represent? By reading the code, they seem to be timestamps, but the variable names don't say that